q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,332 @@
+        """
+        Dispatch the metadata column to be curated to
+        the functions that will be applied to properly process
+        the current variable according to the current set of
+        rules for it.
+
+        Parameters
+        ----------
+        variable : str
+            Name of the variable which rules are parsed and
+            that is the yaml file name without the .yml extension.
+        input_column : pd.Series
+            Metadata column to be curated (that may already got
+            through this function).
+
+        Returns
+        -------
+        output_column : pd.Series
+            Curated (or being curated) metadata column.
+
+        """
+        pass
+
+
+class RulesCollection(object):
+    """Collect all the rules for all the variables.
+
+    Parameters
+    ----------
+    variables_rules_dir : str
+        Path to the folder containing the variables rules in yaml.
+
+    Attributes
+    ----------
+    variables_rules_files : list
+        Paths to the yaml rules files.
+    warnings : WarningsCollection instance
+        Collection of warnings encountered during either the parsing
+        of the variables rules, or during the application the rules.
+    errors : ErrorsCollection instance
+        Collection of errors encountered during either the parsing
+        of the variables rules, or during the application the rules.
+
+    Methods
+    -------
+    check_variables_rules
+        Fills `rules` attribute with correctly formatted rules
+        for the current variable, or raise user-defined exception.
+    _check_variables_rules_dir
+        Performs checks on rules paths input. Either raise Exceptions,
+        or return the list `variables_rules_files` containing the paths
+        to the existing yml rules files.
+    parse_variables_rules
+        Parse the yml rules files, one variable at a time and returns the
+        dict `variables_rules` that for each variable (keys) leads to an
+        instance of the `Rules()` class. This class performs rule-formatting
+        checks and fills a rules dict structure.
+
+    Raises
+    ------
+    IOError
+        If directory passed to --p-rules-dir does not exist.
+    FileNotFoundError
+        If directory passed to --p-rules-dir contains no .yml file.
+
+    """
+
+    def __init__(self) -> None:
+        """Initialize the class instance for the collection of the set
+        of rules associated with all the metadata variables. The instantiated class will be
+        given a `variable` attribute corresponding to the variable name
+        as string, as well as a `rules` attribute collecting the actual
+        rules in a dictionary.
+
+        """
+        self.warnings = WarningsCollection()
+        self.errors = ErrorsCollection()
+        self.variables_rules = {}
+
+    def parse_variables_rules(self, variables_rules_files: list) -> None:
+        """Parse the variables yaml rues files one by one.
```

Comment:
```suggestion
        """Parse the variables yaml rules files one by one.
```

Reply:
accounted (for)[https://www.youtube.com/watch?v=0l6c8tO-E_E] - will be part of upcoming